include/utils.h

```diff
@@ -52,7 +52,8 @@ void extract_filter(const std::string& line,
 std::string retrieve_url(const std::string& url,
 	ConfigContainer* cfgcont = nullptr,
 	const std::string& authinfo = "",
-	const std::string* postdata = nullptr,
+	const std::string* body = nullptr,
+	const std::string& method = "GET",
```

Comment:
Please use an `enum class` to describe methods. That has several advantages over strings:
1. typos turn into compile-time errors;
2. the code becomes self-documenting, since it's now obvious what methods are supported;
3. we can use a `switch` instead of an `if`-ladder, which should make the code more readable (obvious what's going on without "executing it in your head") and further self-documenting (if new variant is added, the code will fail to compile until a corresponding `case` is added to the `switch`).